client/app/src/services/network.service.js

```diff
@@ -326,6 +328,29 @@
       return deferred.promise
     }
 
+    // Returns the BTC value in satoshi
+    function convertToSatoshi(val) {
+        return Number(val).toFixed(8);
+    }
+
+
+    // Updates peer with all currency values relative to the USD price.
+    function updatePeerWithCurrencies(peer, res) {
+        $http.get('https://api.fixer.io/latest?base=USD', { timeout: 2000}).then( function (result) {
+            const USD_PRICE = Number(res.data[0].price_usd)
+            var currencies = ["aud", "brl", "cad", "chf", "cny", "eur", "gbp", "hkd", "idr", "inr", "jpy", "krw", "mxn", "rub"]
+            var prices = {}
+            currencies.forEach(function(currency) {
+                prices[currency] = result.data.rates[currency.toUpperCase()] * USD_PRICE
+            })
+            prices["btc"] = res.data[0].price_btc
+            prices["usd"] = res.data[0].price_usd
+            peer.market.price = prices
```

Comment:
Cache the response here:

```
storageService.set('peerCurrencies', prices, true)
```

Then roughly Line 22 of NetworkService could become:

```
var peer = {
  ip: network.peerseed,
  network: storageService.getContext(),
  isConnected: false,
  height: 0,
  lastConnection: null,
  price: storageService.get('peerCurrencies') || { btc: '0.0' }
}
```

This will need testing. Also worth seeing what the other guys think of this approach

Reply:
setting true in `storageService.set('peerCurrencies', prices, true)` force a full save of the storage into localstorage. If it happens too often the app can be unresponsive. Otherwise the storage is getting flushed to localstorage every 10s or so.

unless this is critical (like before shutting down the app) i would remove this flag.

Reply:
@fix that's good to know! I think this is only called once per network connection but I'll check and make the necessary changes regardless as this isn't critical to force a flush for the prices. This has been merged already as a separate PR so I'll make a quick fix for it. Thanks!